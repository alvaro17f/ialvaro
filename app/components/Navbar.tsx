"use client";
import { Transition } from "@headlessui/react";
import "animate.css";
import Link from "next/link";
import { useEffect, useRef, useState } from "react";

const links = [
	{
		label: "Home",
		route: "/",
	},
	{
		label: "About",
		route: "/about",
	},
];

export default function Navbar() {
	const [isOpen, setIsOpen] = useState(false);
	const ref = useRef(null);

	return (
		<>
			<nav>
				<div className="px-4 mx-auto max-w-7xl sm:px-6 lg:px-8">
					<div className="flex items-center justify-between h-16">
						<div className="flex items-center">
							<div className="flex-shrink-0">
								<Link href="/">
									<img className="w-32" src="azama_full.svg" alt="logo" />
								</Link>
							</div>
							<div className="hidden md:block">
								<div className="flex items-baseline ml-10 space-x-4">
									{links.map(({ label, route }) => (
										<Link
											key={route}
											href={route}
											className="px-3 py-2 text-sm font-medium rounded-md text-azama-text-white hover:bg-azama-primary hover:text-azama-text-dark"
											aria-label={`${label}-desktop`}
										>
											{label}
										</Link>
									))}
								</div>
							</div>
						</div>
						<div className="flex -mr-2 md:hidden">
							<button
								onClick={() => setIsOpen(!isOpen)}
								type="button"
								className="inline-flex items-center justify-center p-2 rounded-md text-azama-text-white focus:outline-none focus:ring-2 focus:ring-offset-2 "
								aria-controls="mobile-menu"
								aria-expanded="false"
							>
								<span className="sr-only">menu-mobile</span>
								{!isOpen ? (
									<svg
										className="block w-6 h-6"
										xmlns="http://www.w3.org/2000/svg"
										fill="none"
										viewBox="0 0 24 24"
										stroke="currentColor"
										aria-hidden="true"
									>
										<path
											strokeLinecap="round"
											strokeLinejoin="round"
											strokeWidth="2"
											d="M4 6h16M4 12h16M4 18h16"
										/>
									</svg>
								) : (
									<svg
										className="block w-6 h-6"
										xmlns="http://www.w3.org/2000/svg"
										fill="none"
										viewBox="0 0 24 24"
										stroke="currentColor"
										aria-hidden="true"
									>
										<path
											strokeLinecap="round"
											strokeLinejoin="round"
											strokeWidth="2"
											d="M6 18L18 6M6 6l12 12"
										/>
									</svg>
								)}
							</button>
						</div>
					</div>
				</div>

				<Transition
					show={isOpen}
					enter="transition ease-out duration-100 transform"
					enterFrom="opacity-0 scale-95"
					enterTo="opacity-100 scale-100"
					leave="transition ease-in duration-75 transform"
					leaveFrom="opacity-100 scale-100"
					leaveTo="opacity-0 scale-95"
				>
					{
						<div className="md:hidden">
							<div ref={ref} className="px-2 pt-2 pb-3 space-y-1 sm:px-3">
								{links.map(({ label, route }) => (
									<Link
										key={route}
										href={route}
										onClick={() => setIsOpen(!isOpen)}
										className="block px-3 py-2 text-base font-medium text-white rounded-md hover:bg-gray-700"
										aria-label={`${label}-mobile`}
									>
										{label}
									</Link>
								))}
							</div>
						</div>
					}
				</Transition>
			</nav>
		</>
	);
}
