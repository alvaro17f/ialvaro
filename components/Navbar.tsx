"use client";
import { motion } from "framer-motion";
import Link from "next/link";
import { useRef, useState } from "react";

const links = [
	{
		label: "Home",
		route: "/",
	},
	{
		label: "Portfolio",
		route: "/portfolio",
	},
	{
		label: "About",
		route: "/about",
	},
];

export default function Navbar() {
	const [isOpen, setIsOpen] = useState(false);
	const ref = useRef(null);

	return (
		<>
			<nav>
				<div className="px-4 mx-auto max-w-7xl sm:px-6 lg:px-8">
					<div className="flex items-center justify-between h-16">
						<div className="flex items-center">
							<div className="flex-shrink-0">
								<Link href="/">
									<img
										width={120}
										height={120}
										src="/images/azama_full.svg"
										alt="logo"
									/>
								</Link>
							</div>
							<div className="hidden md:block">
								<div className="flex items-baseline ml-10 space-x-4">
									{links.map(({ label, route }) => (
										<Link
											key={route}
											href={route}
											className="px-3 py-2 text-sm font-medium rounded-md text-azama-white hover:bg-azama-danger hover:text-azama-dark"
											aria-label={`${label}-desktop`}
										>
											{label}
										</Link>
									))}
								</div>
							</div>
						</div>
						<div className="flex -mr-2 md:hidden">
							<button
								onClick={() => setIsOpen(!isOpen)}
								type="button"
								className="inline-flex items-center justify-center p-2 rounded-md text-azama-white focus:outline-none focus:ring-2 focus:ring-offset-2"
								aria-controls="mobile-menu"
								aria-expanded="false"
							>
								<span className="sr-only">menu-mobile</span>
								{!isOpen ? (
									<svg
										className="block w-6 h-6"
										xmlns="http://www.w3.org/2000/svg"
										fill="none"
										viewBox="0 0 24 24"
										stroke="currentColor"
										aria-hidden="true"
									>
										<path
											strokeLinecap="round"
											strokeLinejoin="round"
											strokeWidth="2"
											d="M4 6h16M4 12h16M4 18h16"
										/>
									</svg>
								) : (
									<svg
										className="block w-6 h-6"
										xmlns="http://www.w3.org/2000/svg"
										fill="none"
										viewBox="0 0 24 24"
										stroke="currentColor"
										aria-hidden="true"
									>
										<path
											strokeLinecap="round"
											strokeLinejoin="round"
											strokeWidth="2"
											d="M6 18L18 6M6 6l12 12"
										/>
									</svg>
								)}
							</button>
						</div>
					</div>
				</div>

				{isOpen && (
					<div ref={ref} className="px-2 pt-2 pb-3 space-y-1 sm:px-3">
						<motion.div
							className="md:hidden"
							initial={{ x: 10, opacity: 0 }}
							animate={{ x: 0, opacity: 1 }}
							exit={{ opacity: 0 }}
							transition={{ duration: 1 }}
						>
							{links.map(({ label, route }) => (
								<Link
									key={route}
									href={route}
									onClick={() => setIsOpen(!isOpen)}
									className="block px-3 py-2 text-base font-medium text-white rounded-md hover:bg-azama-muted"
									aria-label={`${label}-mobile`}
								>
									{label}
								</Link>
							))}
						</motion.div>
					</div>
				)}
			</nav>
		</>
	);
}
