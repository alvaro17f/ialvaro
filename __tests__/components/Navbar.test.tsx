import { afterEach, beforeEach, describe, expect, it } from "vitest";
import {
	render,
	screen,
	within,
	fireEvent,
	cleanup,
} from "@testing-library/react";
import Navbar from "@/app/components/Navbar";

describe("<Navbar />", () => {
	beforeEach(() => {
		render(<Navbar />);
	});
	afterEach(cleanup);

	it("should match the snapshot", () => {
		const { container } = render(<Navbar />);
		expect(container).toMatchSnapshot();
	});

	it("should render the navbar logo", () => {
		const logo = screen.getByAltText("logo");
		expect(logo).toBeTruthy();
	});

	it("should render links", () => {
		const navigation = within(screen.getByRole("navigation"));
		expect(navigation.getByRole("link", { name: /home/i })).toBeDefined();
		expect(navigation.getByRole("link", { name: /about/i })).toBeDefined();
	});

	it("should display the navbar menu when clicking on the mobile menu button", () => {
		const mobileMenuButton = screen.getByRole("button", {
			name: /menu-mobile/i,
		});
		fireEvent.click(mobileMenuButton);
		const menuItems = screen.getAllByRole("link");
		expect(menuItems.length).toBeGreaterThan(0);
		expect(screen.getByLabelText(/home-mobile/i)).toBeDefined();
		expect(screen.getByLabelText(/about-mobile/i)).toBeDefined();
		fireEvent.click(mobileMenuButton);
		expect(screen.queryByLabelText(/about-mobile/i)).toBeNull();
		expect(screen.queryByLabelText(/home-mobile/i)).toBeNull();
	});

	it("should close the navbar menu when a link is clicked", () => {
		const mobileMenuButton = screen.getByRole("button", {
			name: /menu-mobile/i,
		});
		fireEvent.click(mobileMenuButton);
		const menuItems = screen.getAllByRole("link");
		expect(menuItems.length).toBeGreaterThan(0);
		expect(screen.getByLabelText(/home-mobile/i)).toBeDefined();
		expect(screen.getByLabelText(/about-mobile/i)).toBeDefined();
		fireEvent.click(screen.getByLabelText(/home-mobile/i));
		expect(screen.queryByLabelText(/about-mobile/i)).toBeNull();
		expect(screen.queryByLabelText(/home-mobile/i)).toBeNull();
	});
});
