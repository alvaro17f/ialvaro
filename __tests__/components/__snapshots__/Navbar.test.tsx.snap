// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Navbar /> > should match the snapshot 1`] = `
<div>
  <nav
    class="bg-gray-800"
  >
    <div
      class="px-4 mx-auto max-w-7xl sm:px-6 lg:px-8"
    >
      <div
        class="flex items-center justify-between h-16"
      >
        <div
          class="flex items-center"
        >
          <div
            class="flex-shrink-0"
          >
            <a
              href="/"
            >
              <img
                alt="logo"
                class="w-8 h-8"
                src="https://tailwindui.com/img/logos/workflow-mark-indigo-500.svg"
              />
            </a>
          </div>
          <div
            class="hidden md:block"
          >
            <div
              class="flex items-baseline ml-10 space-x-4"
            >
              <a
                class="px-3 py-2 text-sm font-medium text-white rounded-md hover:bg-gray-700"
                href="/"
              >
                Home
              </a>
              <a
                class="px-3 py-2 text-sm font-medium text-white rounded-md hover:bg-gray-700"
                href="/about"
              >
                About
              </a>
            </div>
          </div>
        </div>
        <div
          class="flex -mr-2 md:hidden"
        >
          <button
            aria-controls="mobile-menu"
            aria-expanded="false"
            class="inline-flex items-center justify-center p-2 text-gray-400 bg-gray-900 rounded-md hover:text-white hover:bg-gray-800 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
            type="button"
          >
            <span
              class="sr-only"
            >
              menu-mobile
            </span>
            <svg
              aria-hidden="true"
              class="block w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
  <header
    class="bg-white shadow"
  >
    <div
      class="px-4 py-6 mx-auto max-w-7xl sm:px-6 lg:px-8"
    >
      <h1
        class="text-3xl font-bold text-gray-900"
      />
    </div>
  </header>
</div>
`;
